tasks/set_vars.yml

```
@@ -5,3 +5,7 @@
     gather_subset: min
   when: not ansible_facts.keys() | list |
     intersect(__rhc_required_facts) == __rhc_required_facts
+
+- name: Check if insights-client is installed
+  package_facts:
+  when: rhc_insights.state | d('present') == "absent"
```

```suggestion
  when: rhc_insights.state | d("present") == "absent"
```
please be consistent with quoting - Ansible style prefers double quotes unless the context requires single